macaroon_service.go

```diff
@@ -0,0 +1,364 @@
+package lndclient
+
+import (
+	"context"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"time"
+
+	"github.com/btcsuite/btcd/btcec"
+	"github.com/lightningnetwork/lnd/keychain"
+	"github.com/lightningnetwork/lnd/kvdb"
+	"github.com/lightningnetwork/lnd/lnrpc"
+	"github.com/lightningnetwork/lnd/macaroons"
+	"github.com/lightningnetwork/lnd/rpcperms"
+	"go.etcd.io/bbolt"
+	"google.golang.org/grpc"
+	"gopkg.in/macaroon-bakery.v2/bakery"
+	"gopkg.in/macaroon-bakery.v2/bakery/checkers"
+)
+
+const (
+	// defaultDBName is the default macaroon db name.
+	defaultDBName = "macaroons.db"
+
+	// defaultDBTimeout is the default timeout to be used for the
+	// macaroon db connection.
+	defaultDBTimeout = 5 * time.Second
+)
+
+var (
+	// MacaroonPubKey is the public key point that can be use for when we
+	// are deriving a shared secret key with LND. It was generated via a
+	// try-and-increment approach using the phrase "Shared Secret" with
+	// SHA2-256. The code for the try-and-increment approach can be seen
+	// here: https://github.com/lightninglabs/lightning-node-connect/tree/master/mailbox/numsgen
+	nBytes, _ = hex.DecodeString(
```

Comment:
nit: `sharedKeyNUMSBytes` and split godoc comment? Perhaps the `..Bytes` variable can say how it was generated and the actual pubkey for what it could be used?